x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/cluster/routing/allocation/DataTierAllocationDeciderTests.java

```diff
@@ -483,13 +597,101 @@ public void testPreferredTierAvailable() {
                 DataTierAllocationDecider.preferredAvailableTier(
                     DataTier.parseTierList("data_cold,data_warm,data_hot"),
                     nodes,
-                    updatedDesiredNodes
+                    updatedDesiredNodes,
+                    shutdownMetadata
                 ),
                 equalTo(Optional.of("data_warm"))
             );
         }
     }
 
+    public void testDataTierDeciderConsidersNodeShutdown() {
```

Comment:
can we add a test that goes through the branch where we check that there're still more available nodes in the tier than the number of nodes leaving the cluster?